src/app-layer-frames.c

```diff
@@ -464,23 +473,49 @@ Frame *AppLayerFrameNewByPointer(Flow *f, const StreamSlice *stream_slice,
     return r;
 }
 
+static Frame *AppLayerFrameUdp(Flow *f, const StreamSlice *stream_slice,
+        const uint32_t frame_start_rel, const int64_t len, int dir, uint8_t frame_type)
```

Comment:
should we refuse to create a frame when offset + len exceeds packet payload len?